cylc/flow/suite_db_mgr.py

```diff
@@ -578,8 +580,31 @@ def recover_pub_from_pri(self):
                 f"{self.pri_dao.db_file_name}")
             self.pub_dao.n_tries = 0
 
-    def restart_upgrade(self):
-        """Vacuum/upgrade runtime DB on restart."""
+    def on_restart(self):
+        """Check & vacuum the runtime DB on restart."""
+        if not os.path.isfile(self.pri_path):
+            raise SuiteServiceFileError(
+                'Cannot restart as suite database not found')
+        self.check_suite_db_compatibility()
         pri_dao = self.get_pri_dao()
         pri_dao.vacuum()
         pri_dao.close()
+
+    def check_suite_db_compatibility(self):
+        """Raises SuiteServiceFileError if the existing suite database is
+        incompatible with the current version of Cylc."""
+        pri_dao = self.get_pri_dao()
+        last_run_ver = pri_dao.connect().execute(
+            f'SELECT value FROM {self.TABLE_SUITE_PARAMS} '
+            f'WHERE key == "{self.KEY_CYLC_VERSION}"').fetchone()[0]
```

Comment:
Need to wrap this in a try/except to protect against the failure of this query (the suite_params table is fairly new).